server/server.go

```diff
@@ -893,9 +976,188 @@ func (s *Server) v1QueryGet(w http.ResponseWriter, r *http.Request) {
 	writer.JSON(w, 200, results, pretty)
 }
 
+func constructDiagnosticsInput(r *http.Request) (map[string]interface{}, error) {
```

Comment:
Nit: move helper functions to the bottom instead of interspersing them with the server's methods.